packages/rpc-transport/src/transports/transport-types.ts

```diff
@@ -10,9 +8,10 @@ export interface IRpcTransport {
 }
 
 type RpcWebSocketTransportConfig = Readonly<{
+    payload: unknown;
     signal: AbortSignal;
 }>;
 
 export interface IRpcWebSocketTransport {
-    (config: RpcWebSocketTransportConfig): Promise<RpcWebSocketConnection>;
+    (config: RpcWebSocketTransportConfig): Promise<AsyncIterable<unknown>>;
```

Comment:
Ugh. This isn't going to work. Wrappers like the auto-ping wrapper need to be able to send messages down the _exact same connection_ that they're holding on to an iterator for, so they need a `send()` method to call.

Most of this PR can stay the way it is, but I'll have to expose `send()` on this type. I'll consider whether I need to call it `send_DO_NOT_USE_OR_YOU_WILL_BE_FIRED` or whether misuse is unlikely.

Reply:
Steve never met a `DO_NOT_USE_OR_YOU_WILL_BE_FIRED` he didn't love 😆